src/trade/spread.ts

```diff
@@ -25,16 +30,22 @@ export const getSpreadWithPriceImpactP = (
       getCurrentOiWindowId(oiWindowsSettings),
       oiWindowsSettings.windowsCount,
       oiWindows,
-      buy
+      isOpen === undefined || isOpen ? buy : !buy
     );
   }
 
   if (!onePercentDepth || activeOi === undefined || collateral === undefined) {
-    return pairSpreadP;
+    return pairSpreadP / 2;
   }
 
   return (
-    pairSpreadP +
-    (activeOi + (collateral * leverage) / 2) / onePercentDepth / 100
+    pairSpreadP / 2 +
+    ((activeOi + (collateral * leverage) / 2) / onePercentDepth / 100 / 2) *
+      (protectionCloseFactor === undefined ||
```

Comment:
NIT: Lets export this into another fn that accepts the Context obj I mentioned in the other comment and returns the effective protectionCloseFactor.